### PR TITLE
Add Support for Instructions API

### PR DESCRIPTION
**Pull request by cyberlord29**

### CODE REVIEW

File: lib/kite_connect_ex/instrument.ex [hunk 1]
```diff
@@ -0,0 +1,42 @@
+defmodule KiteConnectEx.Instrument do
+  @moduledoc """
+  Module to represent a user's portfolio
```
Comment (nitinstp23) [inline lib/kite_connect_ex/instrument.ex]: fix this

File: lib/kite_connect_ex/instrument.ex [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  """
+
+  alias KiteConnectEx.Portfolio.Holding
+  alias KiteConnectEx.Request
+
+  @instruments_base_path "/instruments"
+
+  @doc """
+  Get user's portfolio `holdings`
```
Comment (nitinstp23) [inline lib/kite_connect_ex/instrument.ex]: fix this comment

File: lib/kite_connect_ex/instrument.ex [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  ## Example
+
+    {:ok, holdings} = KiteConnectEx.Portfolio.holdings("access-token")
```
Comment (nitinstp23) [inline lib/kite_connect_ex/instrument.ex]: fix!

File: lib/kite_connect_ex/response.ex [hunk 4]
```diff
@@ -7,6 +7,18 @@ defmodule KiteConnectEx.Response do
   @success_status "success"
   @error_status "error"
 
+  @doc false
+  @spec parse_csv_response(HTTPoison.Response.t()) :: success | error
+  def parse_csv_response(%HTTPoison.Response{} = response) do
```
Comment (nitinstp23) [inline lib/kite_connect_ex/response.ex]: how is this different from the `parse_response` function?

File: lib/kite_connect_ex/request.ex [hunk 5]
```diff
@@ -18,6 +18,16 @@ defmodule KiteConnectEx.Request do
   @spec api_version() :: String.t()
   def api_version, do: @api_version
 
+  @spec get_csv(String.t(), map() | nil, list, list) :: Response.success() | Response.error()
+  def get_csv(path, query, headers, opts \\ []) do
```
Comment (nitinstp23) [inline lib/kite_connect_ex/request.ex]: the request module is just a wrapper around all HTTP calls. Can you create a separate module for handling CSV request/response? How are these requests different from regular HTTP requests?

Comment (cyberlord29) [reply lib/kite_connect_ex/request.ex]: It just doesnt use do a Jason decode on the response ,  I was going to convert the csv dump to json but I did in heimdall code instead , where should I be doing this ?  @nitinstp23

File: lib/kite_connect_ex/instrument.ex (hunk 1 above)
Comment (cyberlord29) [reply lib/kite_connect_ex/instrument.ex]: fixed.

File: lib/kite_connect_ex/instrument.ex (hunk 2 above)
Comment (cyberlord29) [reply lib/kite_connect_ex/instrument.ex]: fixed.

File: lib/kite_connect_ex/instrument.ex (hunk 3 above)
Comment (cyberlord29) [reply lib/kite_connect_ex/instrument.ex]: fixed all comments

File: lib/kite_connect_ex/request.ex (hunk 5 above)
Comment (cyberlord29) [reply lib/kite_connect_ex/request.ex]: @nitinstp23 can I pass an ignore optional flag instead so that it doesnt do json decode ?

Comment (nitinstp23) [reply lib/kite_connect_ex/request.ex]: can you pattern match on the response headers in the `KiteConnectEx.Response.parse_response` function?

```elixir
  def parse_response(%HTTPoison.Response{headers: headers} = response) when {"ContentType": "application/json"} in headers do
    # parse JSON response

  def parse_response(%HTTPoison.Response{headers: headers} = response) when {"ContentType": "application/csv"} in headers do
    # parse CSV response
```

**Pull request merged**